clash-ghc/src-ghc/Clash/GHC/NetlistTypes.hs

```diff
@@ -178,7 +178,7 @@ ghcTypeToHWType iw floatSupport = go
             0 -> returnN (Void (Just (BitVector (fromInteger n))))
             _ -> returnN (BitVector (fromInteger n))
 
-        "Clash.Sized.Internal.Index.Index" -> do
+        "Clash.Sized.Internal.Index.SatIndex" -> do
           n <- liftE (tyNatSize m (head args))
           if n < 2
```

Comment:
Now that the size isn't the `head args` anymore, you probably need to do something like:
```haskell
let [_, szTy] = args
n <- liftE (tyNatSize m szTy)
```